src/182B30.c

```diff
@@ -2059,104 +2059,300 @@ void func_8025D160(ActorPart* arg0, s32 arg1) {
     part->decorationTable->unk_8B0[decorationIndex]->flags |= 0x10;
 }
 
-INCLUDE_ASM(s32, "182B30", func_8025D71C);
+void func_8025D71C(ActorPart* part, s32 decorationIndex) {
+    DecorationTable* decor = part->decorationTable;
+    EffectInstance* effect;
+
+    switch (decor->unk_8BC[decorationIndex]) {
+        case 0:
+            decor->unk_8B0[decorationIndex] = fx_65(2, part->currentPos.x, part->currentPos.y, part->currentPos.z, 1.0f, 0);
+            decor->unk_8C6[decorationIndex].unk00 = 1;
+            decor->unk_8BC[decorationIndex] = 1;
+            break;
+        case 1:
+            effect = decor->unk_8B0[decorationIndex];
+            effect->data.unk_65->pos.x = part->currentPos.x;
+            effect->data.unk_65->pos.y = part->currentPos.y;
+            effect->data.unk_65->pos.z = part->currentPos.z;
+            effect->data.unk_65->unk_34 = decor->unk_8C6[decorationIndex].unk00 / 100.0f;
+            break;
+    }
+}
 
 void func_8025D810(ActorPart* part, s32 decorationIndex) {
     part->decorationTable->unk_8B0[decorationIndex]->flags |= 0x10;
 }
 
-INCLUDE_ASM(s32, "182B30", func_8025D830);
+void func_8025D830(ActorPart* part, s32 decorationIndex) {
+    DecorationTable* decor = part->decorationTable;
+    EffectInstance* effect;
+    
+    switch (decor->unk_8BC[decorationIndex]) {
+        case 0:
+            decor->unk_8B0[decorationIndex] = fx_whirlwind(2, part->currentPos.x, part->currentPos.y, part->currentPos.z, 1.0f, 0);
+            decor->unk_8BC[decorationIndex] = 1;
+            break;
+        case 1:
+            effect = decor->unk_8B0[decorationIndex];
+            effect->data.whirlwind->pos.x = part->currentPos.x;
+            effect->data.whirlwind->pos.y = part->currentPos.y;
+            effect->data.whirlwind->pos.z = part->currentPos.z;
+            break;
+    }
+}
 
 void func_8025D8EC(ActorPart* part, s32 decorationIndex) {
     part->decorationTable->unk_8B0[decorationIndex]->flags |= 0x10;
 }
 
-INCLUDE_ASM(s32, "182B30", func_8025D90C);
+void func_8025D90C(ActorPart* part, s32 decorationIndex) {
+    DecorationTable* decor = part->decorationTable;
+    f32 angle, sinA, cosA;
+
+    switch (decor->unk_8BC[decorationIndex]) {
+        case 0:
+            decor->unk_8BE[decorationIndex] = 0;
+            decor->unk_8BC[decorationIndex] = 1;
+            // fall through
+        case 1:
+            decor->unk_8BE[decorationIndex]++;
+            if (decor->unk_8BE[decorationIndex] >= 4) {
+                decor->unk_8BE[decorationIndex] = 0U;
+                angle = (clamp_angle(-part->yaw) * TAU) / 360.0f;
+                sinA = sin_rad(angle);
+                cosA = cos_rad(angle);
+                fx_walking_dust(0,
+                    part->currentPos.x + (part->size.x * sinA * 0.2f),
+                    part->currentPos.y + 1.5f,
+                    part->currentPos.z + (part->size.x * cosA * 0.2f),
+                    sinA, cosA);
+            }
```

Comment:
```suggestion
                }
```

Reply:
It's aligned with the switch, just looks a bit odd since cases are indented.

Reply:
Oh, I see. I got confused by the fx_walking_dust call being indented